events/handler.go

```diff
@@ -144,13 +169,25 @@ func NewRequest() *Request {
 }
 
 // returns a pointer to an initialized watch object
-func newWatch(request *Request, outChannel chan *Event) *watch {
+func newWatch(request *Request, eventChannel *EventChannel) *watch {
 	return &watch{
-		request: request,
-		channel: outChannel,
+		request:      request,
+		eventChannel: eventChannel,
 	}
 }
 
+func (self *events) StopWatch(watch_id int) {
```

Comment:
We can replace this with a Stop() on the EventChannel() that way we don't have to go all the way to the manager to call StopWatch().